### ISSUE TITLE
Very long posts crashes/freezes the mobile app

### ISSUE THREAD

**Issue by paalbra** [mattermost/mattermost-mobile#5956]
#### Summary
Long posts (very many characters) crashes the app on iPhone and Android (not sure about version of Android).

### Environment Information
  - Device Name: iPhone SE
  - OS Version: 15.3
  - Mattermost App Version: 1.48.2 (Build 382)
  - Mattermost Server Version: 5.39.3

#### Steps to reproduce
1. Produce some posts in a channel with a large amount of characters. I created a series of posts close to 2^14/16384 characters (simply the character "x"). Maybe around 10 posts like this.
2. Open the channel in the app

#### Expected behavior
The app shoulden't become unresponsive.

#### Observed behavior (that appears unintentional)
Observe that the app becomes completely unresponsive. If you close and open the app again it will open the same channel and become unresponsive.

The issue seems very similar to what is described in #2828 , but there are no gifs/images here. I've sent large messages with "x" as the only used character.

(The channel was fine in a web browser on PC).

#### Possible fixes
I have no idea. I've tried to spam the channel, to remove the message from recent history. This might provide a kind of workaround. My app is responsive again, but I do not dare to test this furter. I need my app.

**Comment by amyblais** [comment]
@enahum Should I create a ticket for this or do we need more info?

**Comment by josephbaylon** [comment]
@enahum For me, on both Android and iOS, the channel with several long texts (11 posts with max characters each) in it does freeze for about `10 seconds` but eventually loads and the buttons can be interacted with and can add new posts.

Also on Android, switching to another channel freezes the app but eventually loads the other channel and switching back to the long text channel also freezes the app but eventually loads. However, on iOS, switching to another channel is fine but switching back to the long text channel also freezes the app but eventually loads.

In both devices, the app freezes for a bit but doesn't crash.

I tried on both v1.48.2 and v1.49. I have Pixel 3 XL Android 12 and iPhone X iOS 15.3.

**Comment by paalbra** [comment]
PS.
I noticed that the [go library](https://github.com/mattermost/mattermost-server/blob/01a3ed0d3e401d45b52480f1013fd0c0cfc2ba00/model/client4.go#L3585) was getting "invalid post" on _very_ long posts (40k). I then figured that it has a limitation on just below 2^14/16384 characters. I'm not sure if this limitation is in the server or where it is, but that's why I created messages of that length. It seems to be the max length?
